cmd/traffic/cmd/manager/manager.go

```diff
@@ -313,3 +315,19 @@ func (s *service) runSessionGCLoop(ctx context.Context) error {
 		}
 	}
 }
+
+func (s *service) runUpdateTrafficManagerConfigMapLoop(ctx context.Context) error {
+	// Loop updating the agentState every 2mins
+	ticker := time.NewTicker(2 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			err := s.updateTrafficManagerConfigMap(ctx)
```

Comment:
There should be a flag to avoid unnecessary updates. You could for instance add a

```go
  tmConfigMapUpdated atomic.Bool
```
 to the `service` struct